velox/exec/prefixsort/tests/PrefixSortAlgorithmTest.cpp

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <folly/Random.h>
+#include <glog/logging.h>
+#include <gtest/gtest.h>
+
+#include "velox/exec/prefixsort/PrefixSortAlgorithm.h"
+#include "velox/vector/tests/VectorTestUtils.h"
+
+namespace facebook::velox::exec::prefixsort::test {
+
+namespace {
+// Encode logic will use PrefixSort`s encode module API as follow-up.
+#define BSWAP64(x) \
+  ((               \
+      uint64_t)((((uint64_t)(x)&0xff00000000000000ull) >> 56) | (((uint64_t)(x)&0x00ff000000000000ull) >> 40) | (((uint64_t)(x)&0x0000ff0000000000ull) >> 24) | (((uint64_t)(x)&0x000000ff00000000ull) >> 8) | (((uint64_t)(x)&0x00000000ff000000ull) << 8) | (((uint64_t)(x)&0x0000000000ff0000ull) << 24) | (((uint64_t)(x)&0x000000000000ff00ull) << 40) | (((uint64_t)(x)&0x00000000000000ffull) << 56)))
+
+static inline uint8_t flipSign(uint8_t key_byte) {
+  return key_byte ^ 128;
+}
+
+template <typename T>
+void store(const T& val, char* ptr) {
+  memcpy(ptr, (void*)&val, sizeof(val));
+}
+
+inline void encodeData(char* row, int64_t value) {
+  store<uint64_t>(BSWAP64(value), row);
+  row[0] = flipSign(row[0]);
+}
+
+inline void encodeVector(std::vector<long>& vec) {
+  for (auto i = 0; i < vec.size(); i++) {
+    encodeData((char*)vec.data() + i * sizeof(long), vec[i]);
+  }
+}
+} // namespace
+
+class PrefixSortAlgorithmTest : public testing::Test,
+                                public velox::test::VectorTestBase {
+ public:
+  void testQuickSort(size_t size) {
+    // Data1 will be sorted by quickSort.
+    std::vector<long> data1(size);
+    std::generate(
+        data1.begin(), data1.end(), [&]() { return folly::Random::rand64(); });
+    // Data2 will be sorted by std::sort.
+    std::vector<long> data2 = data1;
+    char* start = (char*)data1.data();
+    char* end = start + sizeof(long) * data1.size();
+    uint32_t entrySize = sizeof(long);
+    auto swapBuffer = AlignedBuffer::allocate<char>(entrySize, pool());
+    auto context = PrefixSortContext(entrySize, swapBuffer->asMutable<char>());
+    encodeVector(data1);
+    context.quickSort(
+        start, end, [&](char* a, char* b) { return memcmp(a, b, entrySize); });
+    std::sort(data2.begin(), data2.end());
+    encodeVector(data2);
```

Comment:
nit: it would be nice to provide decode function and use it to decode results after prefix-sort, then compare with std::sort results directly. Otherwise, if there is a bug in encode (let's say everything is encoding into a constant value), the test will pass.